coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorExecutorImpl.java

```diff
@@ -31,17 +31,7 @@
 import java.util.concurrent.RejectedExecutionException;
 
 public class CoordinatorExecutorImpl<S extends CoordinatorShard<U>, U> implements CoordinatorExecutor<U> {
-    private static class TaskResult<R> {
-        final R result;
-        final Throwable exception;
-
-        TaskResult(
-            R result,
-            Throwable exception
-        ) {
-            this.result = result;
-            this.exception = exception;
-        }
+    private record TaskResult<R>(R result, Throwable exception) {
     }
```

Comment:
Could we merge them into one line?
```suggestion
    private record TaskResult<R>(R result, Throwable exception) { }
```